server/service/integration_core_test.go

```diff
@@ -2113,11 +2113,36 @@ func (s *integrationTestSuite) TestHostDeviceMapping() {
 	s.DoJSON("GET", fmt.Sprintf("/api/latest/fleet/hosts/%d/device_mapping", hosts[1].ID), nil, http.StatusOK, &listResp)
 	require.Len(t, listResp.DeviceMapping, 0)
 
-	// search host by email address finds the corresponding host
 	var listHosts listHostsResponse
+	// list hosts response includes device mappings
+	s.DoJSON("GET", "/api/latest/fleet/hosts", nil, http.StatusOK, &listHosts)
+	require.Len(t, listHosts.Hosts, 3)
+	hostsByID := make(map[uint]HostResponse)
+	for _, h := range listHosts.Hosts {
+		hostsByID[h.ID] = h
+	}
+	var dm []*fleet.HostDeviceMapping
+
+	// device mapping for host 1
+	err := json.Unmarshal(*hostsByID[1].DeviceMapping, &dm)
```

Comment:
host 1 is `hostsByID[0]`

Reply:
Thank you for double checking! The references is to a map so I don't think we have an off-by-one problem so I don't think we have an off-by-one problem in this case at least :)